app/models/presenters/batch_submenu_presenter.rb

```diff
@@ -56,6 +56,19 @@ def build_submenu
       add_submenu_option pluralize(@batch.comments.size, 'comment'), batch_comments_path(@batch)
       load_pipeline_options
       add_submenu_option 'NPG run data', "#{configatron.run_data_by_batch_id_url}#{@batch.id}"
+      return unless aviti_run_manifest?
+      add_submenu_option 'Download Sample Sheet', id: @batch.id, controller: :batches, action: :generate_sample_sheet
+    end
+
+    # This is used to determine if we need to display the Aviti run manifest option
+    # in the batch submenu.
+    # @return [Boolean] true if the batch is released and has Element Aviti requests
+    def aviti_run_manifest?
+      @batch.released? && aviti_requests?
+    end
+
+    def aviti_requests?
+      @batch.requests.any? { ElementAvitiSequencingRequest }
```

Comment:
How does this work? Is this checking the type of the request or request class?

Reply:
It is not the same as `is_a?` . It means, check for any requests, the block returns true; truthy; other than nil or false.